src/CalendarComponent.jsx

```diff
@@ -96,7 +104,7 @@ class CalendarComponent extends React.Component {
 				id={id}
 				type='text'
 				name={name}
-				defaultValue={this.state.value}
+				defaultValue={this.updateDate()}
```

Comment:
It feels a little strange to update a date at the same time that you are assigning a default - those seem like two separate operations. In some cases, the function also returns `undefined` as the default - is that expected?